### PR TITLE
Feature/ap5555 api spring boot migration

### PR DESCRIPTION
**Pull request by jamesjin2021**
This is a PR for Springboot migration for existing REST-Endpoint module (Jersey based). Some changes are introduced to ApromoreCore to allow API requests/Swagger requests pass through and enhance backend services.

This PR is a dependent of another ApromoreEE PR https://github.com/apromore/ApromoreEE/pull/998, this means that 2 PRs need to be merged at the same time.

### CODE REVIEW

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/config/PortalKeyCloakSecurity.java [hunk 1]
```diff
@@ -87,7 +103,9 @@ protected void configure(HttpSecurity http) throws Exception {
 
     http.headers().frameOptions().sameOrigin();
     http.addFilterAfter(new SameSiteFilter(), BasicAuthenticationFilter.class);
-    http.csrf().ignoringAntMatchers("/zkau", "/rest", "/rest/*", "/rest/**/*", "/zkau/*", "/bpmneditor/editor/*").and()
+    http.csrf().ignoringAntMatchers("/zkau", "/rest", "/rest/*", "/rest/**/*", "/zkau/*", "/bpmneditor/editor/*")
```
Comment (lahiru-maha) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/config/PortalKeyCloakSecurity.java:92]: Is it worth externalising these paths to be read from application config?

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/config/PortalSecurityConfig.java [hunk 2]
```diff
@@ -34,6 +34,21 @@
 @ConditionalOnProperty(prefix = "keycloak", name = "enabled", havingValue = "false")
 public class PortalSecurityConfig extends WebSecurityConfigurerAdapter {
 
+  private static final String[] SWAGGER2_AUTH_WHITELIST = {
```
Comment (lahiru-maha) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/config/PortalSecurityConfig.java]: Seems like PortalSecurityConfig and PortalKeyCloakSecurity both share the same whitelists? Would be worth adding to a common whitelist constants, or define in an externalised application configuration.

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/security/impl/UsernamePasswordAuthenticationProvider.java [hunk 3]
```diff
@@ -52,10 +52,7 @@
 import javax.inject.Inject;
 import javax.persistence.NoResultException;
 import java.security.NoSuchAlgorithmException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (lahiru-maha) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/security/impl/UsernamePasswordAuthenticationProvider.java:55]: Minor one, but its actually better to have the individual imports than *

=> yeah, this is a Checkstyle issue, but it is in another module 'Apromore-Manager', I prefer to defer it for future.

File: Apromore-Database/src/main/resources/db/migration/changeLog-Data-RestAPI.yaml [hunk 4]
```diff
@@ -0,0 +1,622 @@
+databaseChangeLog:
+- changeSet:
+    id:  91220115105200
+    author:  "add new test user to user table"
```
Comment (raboczi) [inline Apromore-Database/src/main/resources/db/migration/changeLog-Data-RestAPI.yaml]: In all these author fields, you are the author.


=> Author fields have been updated to "James Jin <james.jin@apromore.com>" in test data file 'changeLog-Data-RestAPI.yaml' in 'REST-API\src\test\resources'

File: Apromore-Database/src/main/resources/db/migration/changeLog-RestAPI.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+databaseChangeLog:
+   - include:
+       file: db/migration/changeLog-Schema-init.yaml
+
+   - changeSet:
+       id: 1599618934768-128  
+       author: init-data
```
Comment (raboczi) [inline Apromore-Database/src/main/resources/db/migration/changeLog-RestAPI.yaml]: In all these author fields, you are the author.

**Review by raboczi — CHANGES_REQUESTED** [review]
I'm pretty sure all the HTTP error pages in Apromore-Core-Components/Apromore-Portal/src/main/webapp/pages/ (e.g. 404.html, 500.html) ought to be .zul files instead.  Surely that raw ZK Javascript wasn't written by hand.

=> yeah, these HTTP error pages are translated from .ZUL files.  They were used by API Global controller advice to handle both HTML requests and API requests. I just rested the UI pages and APIs, they seem not necessary now,  I have removed them from 'Apromore-Portal' and refactored API Global controller advice accordingly.

**Comment by jamesjin2021** [comment]
> I'm pretty sure all the HTTP error pages in Apromore-Core-Components/Apromore-Portal/src/main/webapp/pages/ (e.g. 404.html, 500.html) ought to be .zul files instead. Surely that raw ZK Javascript wasn't written by hand.

Author fields have been updated.

**Review by raboczi — APPROVED** [review]
This is half the size it started out as, which is great.

**Comment by jamesjin2021** [comment]
looks good. remaining issues would be tracked by Jira cards.

**Comment by jamesjin2021** [comment]
code looks good, merge now.

**Pull request merged**